File: functions/fsharp/NestedFunctions.fs
module NestedFunctions

let printIfNumberIsEvenOrOdd number =
    let verifyIfNumberIsEven = number % 2 = 0
    let printEvenNumber() = printfn "The number %i is even" number
    let printOddNumber() = printfn "The number %i is odd" number
    
    if verifyIfNumberIsEven then
        printEvenNumber()
    else
        printOddNumber()

File: functions/csharp/NestedFunctions.cs
using System;

namespace csharp
{
    static class NestedFunctions
    {
        public static void PrintIfNumberIsEvenOrOdd(int number)
        {
            Func<bool> verifyIfNumberIsEven = () => number % 2 == 0;

            Action printEvenNumber =
                () => Console.WriteLine(string.Format("The number {0} is even", number));

            Action printOddNumber =
                () => Console.WriteLine(string.Format("The number {0} is odd", number));

            if(verifyIfNumberIsEven())
                printEvenNumber();
            else
                printOddNumber();
        }
    }
}